cpp/src/gandiva/filter.h

```diff
@@ -15,8 +15,7 @@
 // specific language governing permissions and limitations
 // under the License.
 
-#ifndef GANDIVA_EXPR_FILTER_H
-#define GANDIVA_EXPR_FILTER_H
+#pragma once
```

Comment:
What is the preferred style? `#pragma once` or the traditional guards?

Reply:
Personally I'd prefer to move to all `#pragma once`. This is well supported in all the compilers we use. I changed it here because the include guard name was out of sync with the file path